quarkus.jdt.ext/com.redhat.microprofile.jdt.quarkus.test/src/main/java/com/redhat/microprofile/jdt/quarkus/renarde/RenardeJaxRsTest.java

```diff
@@ -0,0 +1,125 @@
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.microprofile.jdt.quarkus.renarde;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.lsp4j.CodeLens;
+import org.eclipse.lsp4j.Command;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.SymbolInformation;
+import org.eclipse.lsp4j.SymbolKind;
+import org.eclipse.lsp4mp.commons.MicroProfileJavaCodeLensParams;
+import org.eclipse.lsp4mp.jdt.core.BasePropertiesManagerTest;
+import org.eclipse.lsp4mp.jdt.core.PropertiesManagerForJava;
+import org.eclipse.lsp4mp.jdt.core.utils.IJDTUtils;
+import org.eclipse.lsp4mp.jdt.core.utils.JDTMicroProfileUtils;
+import org.junit.Test;
+
+/**
+ * Tests for the CodeLens features introduced by {@link com.redhat.microprofile.jdt.internal.quarkus.renarde.java.RenardeJaxRsInfoProvider}.
+ */
+public class RenardeJaxRsTest extends BasePropertiesManagerTest {
+
+	private static String quarkus_renarde_todo = "quarkus-renarde-todo";
+
+	@Test
+	public void codeLens() throws Exception {
+		IJavaProject javaProject = loadMavenProject(quarkus_renarde_todo);
+
+		assertNotNull(javaProject);
+
+		MicroProfileJavaCodeLensParams params = new MicroProfileJavaCodeLensParams();
+		params.setCheckServerAvailable(false);
+		IFile javaFile = javaProject.getProject()
+				.getFile(new Path("src/main/java/rest/Application.java"));
+		params.setUri(javaFile.getLocation().toFile().toURI().toString());
+		params.setUrlCodeLensEnabled(true);
+
+		assertCodeLens(params, JDT_UTILS, //
+				ca("http://localhost:8080/", "", r(14, 20, 14, 20)), //
+				ca("http://localhost:8080/about", "", r(19, 25, 19, 25)), //
+				ca("http://localhost:8080/Application/test", "", r(9, 30, 9, 30)));
+	}
+
+	@Test
+	public void workspaceSymbols() throws Exception {
+		IJavaProject javaProject = loadMavenProject(quarkus_renarde_todo);
+
+		assertNotNull(javaProject);
+
+		assertWorkspaceSymbol(javaProject, JDT_UTILS, //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""), //
+				si(""));
+	}
+
+	private static void assertCodeLens(MicroProfileJavaCodeLensParams params, IJDTUtils utils, CodeLens... expected) throws JavaModelException {
```

Comment:
This method should be moved to MicroProfileForJavaAssert from lsp4mp and it should be nice that test on LSP4MP uses this method for codelens.